Automattic/jetpack
# CRM: Introduce Jetpack Autoloader and declare Dompdf in Composer

We used to have a "hardcoded" version of `Dompdf` in our repository which doesn't play nice with some of Jetpack Monorepo's `.gitignore` rules.

This PR aim to correct that by implementing `Dompdf` via Composer instead and autoload the package when needed.

## Proposed changes:

* Introduce `Jetpack Autoloader`.
* Define Dompdf dependency in composer.
* Refactor existing code to prevent `require()` attempts for the old location.
* Add a deprecated tag to `ZeroBSCRM::checkDompdfVersion`.
  * It shouldn't be necessary to do so due to the `ZeroBSCRM` class being `final` but we'll remove it with a refactor where we remove the entire library system.
* Remove all files in `projects/plugins/crm/includes/lib/`  since we no longer use them.

### Other information:


## Jetpack product discussion



p1673639270829519-slack-C04J4UG3FLG

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up a version of this branch 
* Go to `/wp-admin/admin.php?page=manage-customers` and add a Contact
  *  You can use an existing contact as well.
* Go to the "New Quote" page (`/wp-admin/admin.php?page=zbs-add-edit&action=edit&zbstype=quote`)
* Create a new quote
* Press the `Download PDF` button at the bottom of the page
* Verify that the quote looks correct (basically: that it's generated and it isn't blank)
